lib/services/push_notifications/push_notifications.dart

```diff
@@ -70,8 +95,13 @@ class PushNotificationsService {
   }
 
   void promptUserForPushNotificationPermission() async {
-    OneSignal.shared
+    bool hasAllowed = await OneSignal.shared
         .promptUserForPushNotificationPermission(fallbackToSettings: true);
+    if (hasAllowed) {
+      enablePushNotifications();
+    } else {
+      disablePushNotifications();
+    }
   }
```

Comment:
Currently, lineks 100-104 are never called, this will be fixed once OneSignalSDK fixes that issue.